typescript/packages/core/src/plugins/send-nft.ts

```diff
@@ -0,0 +1,64 @@
+import { Connection, PublicKey } from "@solana/web3.js";
+import { parseUnits } from "viem";
+import type { SolanaWalletClient } from "../wallets";
+import type { Plugin } from "./plugins";
+import { z } from "zod";
+import { SystemProgram } from "@solana/web3.js";
+import {
+	mplBubblegum,
+	getAssetWithProof,
+	transfer,
+} from "@metaplex-foundation/mpl-bubblegum";
+import { createUmi } from "@metaplex-foundation/umi-bundle-defaults";
+import {
+	fromWeb3JsPublicKey,
+	toWeb3JsInstruction,
+} from "@metaplex-foundation/umi-web3js-adapters";
+
+export function transferNFT(): Plugin<SolanaWalletClient> {
+	return {
+		name: "transfer_nft",
+		supportsSmartWallets: () => true,
+		supportsChain: (chain) => chain.type === "solana",
+		getTools: async () => {
+			return [
+				{
+					name: "transfer_nft",
+					description:
+						"This {{tool}} sends an NFT from your wallet to an address on a Solana chain.",
+					parameters: transferNFTParametersSchema,
+					method: transferNFTMethod,
+				},
+			];
+		},
+	};
+}
+
+const transferNFTParametersSchema = z.object({
+	recipientAddress: z.string().describe("The address to send the NFT to"),
+	assetId: z.string().describe("The asset ID of the NFT to send"),
+});
+
+async function transferNFTMethod(
+	walletClient: SolanaWalletClient,
+	parameters: z.infer<typeof transferNFTParametersSchema>,
+): Promise<string> {
+	const { recipientAddress, assetId } = parameters;
+	const umi = createUmi(walletClient.connection);
```

Comment:
hmm do we need to expose connection here? what is `createUmi` doing?

Is there a way we can do this without it?

Reply:
Not really, to interact with cNFTs you have to pull a bunch of data from the chain since they make heavy use of merkle trees

Reply:
Not sure, I don't think we can rule out other programs using it though